### PR TITLE
Improved user flow

### PR DESCRIPTION
**Pull request by Sauloxd**
As we talked, I improved the verification step to add a "check app has dependencies"!
Also, I removed and refactored all that was not being used!

Will provide a deployed app so you can test again, but If possible, you can already start checking the code out!

### CODE REVIEW

File: src/applications-guide/applications-guide.component.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  justify-content: space-between;
+}
+
+& .next-step {
+  margin-top: 1.6rem;
+}
+`;
+
+function CollapsableContent({ title, children }) {
+  const [visible, setVisible] = useState(false);
```
Comment (filoxo) [inline src/applications-guide/applications-guide.component.js:116]: We could avoid even needing a CollapsableContent component if we use the [HTML details & summary](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) elements.

File: src/shared/registered-app.component.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+  const { removeApplication } = useContext(LocalStorageContext);
+
+  return (
+    <div {...scope}>
+      <div className="registered-app">
+        <Code code={indentedCode(app)} />
+        {interactive && (
+          <>
+            <div role="button" tabIndex={0} onClick={edit} className="edit">
```
Comment (filoxo) [inline src/shared/registered-app.component.js:18]: Could these become proper `button`s instead? Divs with onClick handlers are inaccessible because a keyboard user cannot interact with it.

File: src/applications-guide/applications-guide.component.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+2. Go to https://single-spa-playground.org/playground/instant-test?name=@org/app&url=8500 to see it working!
+          `}
+          />
+          <CollapsableContent title="The first was to start your webpack-dev-server using https">
+            <>
+              <h3>Do I need https?</h3>
+            </>
+            <p>
+              No, not at all! The only reason we ask you to run https is so that
+              this playground (which is also on https) can download your assets
```
[the remark this answers is not in the document]
Comment (joeldenning) [reply src/applications-guide/applications-guide.component.js:69]: Additionally, the easiest thing to do for HTTPS localhost is to change a chrome setting so you never have to worry about it. We should encourage them to do so: https://stackoverflow.com/a/31900210/1958941

File: src/root-config-guide/root-config-guide.component.js [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+          </a>{" "}
+          What is important for now, is this line:
+        </p>
+        <Code code={`System.register(["react","react-dom"], function(...`} />
+        <p>
+          This is how systemjs manages dependencies for systemjs modules. So,
+          before running your single-spa react application, systemjs need to
+          know what to do with "react" and "react-dom"! Just add them to your
+          root-config import-map and you are ready to go! Also, don't forget to
+          add both root-config and single-spa application!
```
Comment (joeldenning) [inline src/root-config-guide/root-config-guide.component.js:131]: ****
```suggestion
          add both the root-config and single-spa application!
```

File: src/root-config-guide/root-config-guide.component.js [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+          <button
+            type="button"
+            className="primary"
+            onClick={() => {
+              setCopying(true);
+              copy(code);
+            }}
+            disabled={copying}
+          >
+            Copy import-map
```
Comment (joeldenning) [inline src/root-config-guide/root-config-guide.component.js:144]:
```suggestion
            Copy import map
```

File: src/root-config-guide/root-config-guide.component.js [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+  }, [copying]);
+  const { application } = useContext(LocalStorageContext);
+  const appUrls = window.importMapOverrides.getOverrideMap().imports;
+  if (!application)
+    return <p>You need to register at least one application!</p>;
+  const code = `
+<script type="systemjs-importmap">
+  {
+    "imports": {
+      "react": "https://cdn.jsdelivr.net/npm/react@16.13.0/umd/react.production.min.js",
```
Comment (joeldenning) [inline src/root-config-guide/root-config-guide.component.js:28]: We shouldn't hard code react in here - single-spa-vue receives more traffic than single-spa-react does. Instead we should base the import map on the framework that they are using. This is why in the existing playground we ask them what framework they are using.

File: src/applications-guide/applications-guide.component.js (hunk 1 above)
Comment (Sauloxd) [reply src/applications-guide/applications-guide.component.js:116]: Didn't know these existed at all! Thanks!

File: src/applications-guide/applications-guide.component.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Sauloxd) [reply src/applications-guide/applications-guide.component.js:69]: I vote we remove HTTPS from both playground and create-single-spa, but encourage them in more in-depth guides, maybe a section on a "how to use importmap-override" (I think it already is) should be enough, no need to encourage this early on the learning stages

File: src/root.component.js [hunk 7]
```diff
@@ -32,7 +32,7 @@ export default function Root(props) {
 
   useEffect(() => {
     if (
-      !localStorage.getItem("single-spa-playground") &&
+      !localStorage.getItem("single-spa-playground").application &&
```
Comment (joeldenning) [inline src/root.component.js]: localStorage.getItem() returns a string, not an object. Which means that `str.application` will always be falsy? I think you need to JSON.parse() this before checking for the application property

File: src/instant-test/instant-test.component.js [hunk 8]
```diff
@@ -4,34 +4,24 @@ import { LocalStorageContext } from "../shared/use-local-storage-data.hook";
...
-    resetApplications,
-  } = React.useContext(LocalStorageContext);
+  const { addApplication } = React.useContext(LocalStorageContext);
 
   React.useEffect(() => {
     if (!isValid) {
       return;
     }
 
-    resetApplications();
```
Comment (joeldenning) [inline src/instant-test/instant-test.component.js:19]: Why was this removed? This is an important feature so that if you use the instant-test route twice that you don't accidentally keep around the previous app that you were testing which causes confusion when you're using it.

File: src/root-config-guide/root-config-guide.component.js [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+        <div>
+          <Code code={`npx create-single-spa --module-type root-config`} />
+        </div>
+        <p>
+          This command won't give you a running root-config out of the box for
+          our playground demonstration! But, fear not, you will understand
+          everything root-config related now!
+        </p>
+        <p>
+          First, lets talk about the{" "}
```
Comment (joeldenning) [inline src/root-config-guide/root-config-guide.component.js:80]: Should either be `first, let's talk about import maps` or `first, let's talk about the import map`

```suggestion
          First, let's talk about {" "}
```

File: src/root-config-guide/root-config-guide.component.js [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+                application.name
+              ]
+            }
+            target="_blank"
+          >
+            JS generated by webpack now?
+          </a>{" "}
+          What is important for now, is this line:
+        </p>
+        <Code code={`System.register(["react","react-dom"], function(...`} />
```
Comment (joeldenning) [inline src/root-config-guide/root-config-guide.component.js:124]: Earlier in the tutorial, it says this would have a "react focus." But also we have tried to extract the framework by running a regex on the code - could we make this more specific to the framework they're using, if we know it?

File: src/root-config-guide/root-config-guide.component.js [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+          And that's all for now! Feedbacks would be appreciated,{" "}
+          <SlackLink>join us in our slack</SlackLink> and give us a hello!
+        </p>
+      </article>
+    </>
+  );
+}
+
+function copy(str) {
+  const el = document.createElement("textarea");
```
Comment (joeldenning) [inline src/root-config-guide/root-config-guide.component.js:175]: small trick I learned recently:

```js
const el = Object.assign(document.createElement('textarea'), {
  value: str,
  readonly: true,
})
```

Unfortunately you can't do this with style.position or style.left - the style property on dom elements doesn't work with Object.assign. But you can for other properties.

File: src/verify-app-guide/verification-steps/application-dependencies.component.js [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+      })
+      .catch((err) => {
+        setError(err);
+        console.error(err);
+        throw err;
+      });
+  }
+});
+
+export const getAppDependencies = ({ name, url }) => {
```
Comment (joeldenning) [inline src/verify-app-guide/verification-steps/application-dependencies.component.js:90]: Wow this is quite fancy. It works! Impressive

File: src/shared/links.component.js [hunk 13]
```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import { navigateToUrl } from "single-spa";
+
+export const SlackLink = ({ children }) => (
+  <a
+    target="_blank"
```
Comment (joeldenning) [inline src/shared/links.component.js:6]: `rel="nooopener"` is generally a good practice for links to external websites. https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/noopener

File: src/shared/registered-app.component.js [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+    <div {...scope}>
+      <div className="registered-app">
+        <Code code={indentedCode(app, importMap)} />
+        {interactive && (
+          <>
+            <button type="button" onClick={edit} className="edit">
+              Edit
+            </button>
+            <button type="button" onClick={removeButton} className="remove">
+              Remove
```
Comment (joeldenning) [inline src/shared/registered-app.component.js:27]: Now that exactly one application is required, what does removing that application do? Does it put the user into an invalid state?

File: src/verify-app-guide/verification-steps/application-executable.component.js [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+        <div className="step-summary">
+          Application code executes without a javascript error
+        </div>
+      </div>
+      {(isExpanded || error) && (
+        <div className="step-detail">
+          <CodeOutput code={error} />
+          {unableToResolveBareSpecifier && (
+            <div>
+              This probably means your application is trying to System.import( '
```
Comment (joeldenning) [inline src/verify-app-guide/verification-steps/application-executable.component.js:75]:
```suggestion
              This probably means your application is trying to System.import('
```

File: src/verify-app-guide/verification-steps/application-executable.component.js [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+              {unableToResolveBareSpecifier}') but it's not defined in import
```
Comment (joeldenning) [inline src/verify-app-guide/verification-steps/application-executable.component.js:76]:
```suggestion
              {unableToResolveBareSpecifier}') but it's not defined in the import
```

File: src/verify-app-guide/verify-app-guide.component.js [hunk 17]
```diff
@@ -0,0 +1,157 @@
...
+    window.importMapOverrides.addOverride(app.name, url);
+    app.url = url;
+    addApplication(app);
+    window.location.reload();
+    setShowRegisterApp(false);
+  }
+}
+
+const css = `
+  &
```
Comment (joeldenning) [inline src/verify-app-guide/verify-app-guide.component.js:156]: Let's just remove the entire `css` variable since it's unused

File: src/applications-guide/applications-guide.component.js [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+          <p>
+            The second is a redirect to this playground, so we can bootstrap
+            your single-spa application for you! And there it is! You have a
+            single-spa application, being served by your localhost, and being
+            run in a deployed application!
+          </p>
+          <p>Welcome to the microfrontend era!</p>
+          <p>
+            Before we guide you into making the{" "}
+            <a href="https://single-spa.js.org/docs/configuration">
```
Comment (joeldenning) [inline src/applications-guide/applications-guide.component.js:64]: maybe in a new tab?

```suggestion
            <a href="https://single-spa.js.org/docs/configuration" target="_blank" rel="noopener">
```

File: src/introduction/introduction.component.js [hunk 19]
```diff
@@ -29,49 +34,41 @@ export default function Introduction(props) {
           applications.
         </section>
         <section>
-          The playground isn't the only way to use single-spa, but is one of the
-          easier ways. It will set you up with single-spa,{" "}
-          <a href="https://github.com/systemjs/systemjs">systemjs@6</a>, and{" "}
-          <a href="https://github.com/WICG/import-maps">import maps</a>. This
-          lets you do the following:
+          The playground will guide you to build a successful singe-spa
```
Comment (joeldenning) [inline src/introduction/introduction.component.js:37]:
```suggestion
          The playground will guide you to build a successful single-spa
```

File: src/introduction/introduction.component.js [hunk 20]
```diff
@@ -29,49 +34,41 @@ export default function Introduction(props) {
...
+          application, highlighting important steps and give tools to help you
```
Comment (joeldenning) [inline src/introduction/introduction.component.js:38]:
```suggestion
          application, highlighting important steps, and giving you tools to help
```

File: src/introduction/introduction.component.js [hunk 21]
```diff
@@ -29,49 +34,41 @@ export default function Introduction(props) {
...
+          understand what went wrong. Mind that this isn't the only way to use
+          single-spa, but by better understanding the concepts, you'll be able
```
Comment (joeldenning) [inline src/introduction/introduction.component.js:40]:
```suggestion
          single-spa, but by better understanding the concepts you'll be able
```

File: src/introduction/introduction.component.js [hunk 22]
```diff
@@ -29,49 +34,41 @@ export default function Introduction(props) {
...
+          to apply it in any way you like. Here's what we'll do:
           <ol>
-            <li>Have microfrontends that are in separate git repos.</li>
             <li>
-              Use different frameworks in the microfrontends, if you wish.
+              Have micro-frontends that are in separate git repos. We will be
+              focusing on React, but you could use Angular, Vue, etc
             </li>
-            <li>Deploy your microfrontends independently.</li>
+            <li>Test if your microfrontend is build up correctly</li>
```
Comment (joeldenning) [inline src/introduction/introduction.component.js:47]:
```suggestion
            <li>Test if your microfrontend is built correctly</li>
```

File: src/introduction/introduction.component.js [hunk 23]
```diff
@@ -29,49 +34,41 @@ export default function Introduction(props) {
...
             <li>
-              Not have to worry about some of the initial setup and
-              configuration.
+              Understand and create a{" "}
+              <a href="https://single-spa.js.org/docs/configuration">
```
Comment (joeldenning) [inline src/introduction/introduction.component.js:50]:
```suggestion
              <a href="https://single-spa.js.org/docs/configuration" target="_blank" rel="noopener">
```

File: src/introduction/introduction.component.js [hunk 24]
```diff
@@ -29,49 +34,41 @@ export default function Introduction(props) {
...
+                root config
+              </a>
+              , which will do the routing of your micro-frontends
```
Comment (joeldenning) [inline src/introduction/introduction.component.js:53]:
```suggestion
              , which will do the routing for your microfrontends.
```

File: src/introduction/introduction.component.js [hunk 25]
```diff
@@ -29,49 +34,41 @@ export default function Introduction(props) {
...
             </li>
+            <li>Deploy your micro-frontends independently.(coming soon)</li>
```
Comment (joeldenning) [inline src/introduction/introduction.component.js:55]: Are you planning on working on this within the playground?

Perhaps for now we could just link to [these docs](https://single-spa.js.org/docs/recommended-setup#deployment-and-continuous-integration-ci)?

File: src/root-config-guide/root-config-guide.component.js [hunk 26]
```diff
@@ -0,0 +1,192 @@
...
+          is actually the origin of the single-spa name (only a "single"
+          single-page-application -- don't think about it too hard{" "}
+          <span role="img" aria-label="Smiley emoji">
+            😀
+          </span>
+          )
+        </section>
+        <section>
+          The single-spa root config is something that we can provide for you
+          entirely. Let's use the create-single-spa again!
```
Comment (joeldenning) [inline src/root-config-guide/root-config-guide.component.js:63]:
```suggestion
          entirely. Let's use create-single-spa again!
```

File: src/root-config-guide/root-config-guide.component.js [hunk 27]
```diff
@@ -0,0 +1,192 @@
...
+        <div>
+          You'll want to create an{" "}
+          <span className="whole-new-git-repo">entirely new git repo</span> for
+          your single-spa root config, separated from your applications.
+        </div>
+        <div>
+          <Code code={`npx create-single-spa --module-type root-config`} />
+        </div>
+        <p>
+          This command won't give you a running root-config out of the box for
```
Comment (joeldenning) [inline src/root-config-guide/root-config-guide.component.js:75]: Could you clarify what you mean by this? In what way does create-single-spa not give you a valid root config? Just that they need to add their application and import map to it?

Maybe clarify this language

File: src/root-config-guide/root-config-guide.component.js [hunk 28]
```diff
@@ -0,0 +1,192 @@
...
+          our playground demonstration! But, fear not, you will understand
```
Comment (joeldenning) [inline src/root-config-guide/root-config-guide.component.js:76]:
```suggestion
          our playground demonstration! But fear not, you will understand
```

File: src/root-config-guide/root-config-guide.component.js [hunk 29]
```diff
@@ -0,0 +1,192 @@
...
+            src="https://www.youtube.com/embed/Lfm2Ge_RUxs"
+            frameBorder="0"
+            allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
+            allowFullScreen
+          ></iframe>
+        </div>
+        <p>
+          Import maps are a browser specification for controlling which{" "}
+          <b>URL to download javascript bundles from</b>. The npm library{" "}
+          <a href="https://github.com/systemjs/systemjs" target="_blank">
```
Comment (joeldenning) [inline src/root-config-guide/root-config-guide.component.js:100]:
```suggestion
          <a href="https://github.com/systemjs/systemjs" target="_blank" rel="noopener">
```

File: src/root-config-guide/root-config-guide.component.js [hunk 30]
```diff
@@ -0,0 +1,192 @@
...
+          But how does my application runs in the playground? Well, for your
+          convenience, we already added "react" and "react-dom" in our import
+          map! Just inspect the DOM and see for yourself!
+        </p>
+
+        <p>Now we need to start the root-config. Run:</p>
+        <Code code={`yarn start`} />
+        <p>
+          But wait... Why am I only seeing a blank page? Well, did you
+          registered you single-spa application? You can add them to your import
```
Comment (joeldenning) [inline src/root-config-guide/root-config-guide.component.js:156]:
```suggestion
          register your single-spa application? You can add them to your import
```

File: src/root-config-guide/root-config-guide.component.js [hunk 31]
```diff
@@ -0,0 +1,192 @@
...
+          map, but if you never register them in single-spa{" "}
```
Comment (joeldenning) [inline src/root-config-guide/root-config-guide.component.js:157]:
```suggestion
          map, but if you never register them via{" "}
```

File: src/root-config-guide/root-config-guide.component.js [hunk 32]
```diff
@@ -0,0 +1,192 @@
...
+          <a href="https://single-spa.js.org/docs/api.html#registerapplication">
+            registerApplication
+          </a>
+          , it will never bootstrap your applications! Just add this code in you
+          root-config entrypoint file.
+        </p>
+        <Code code={registerAppCode} />
+        <p>
+          And that's all for now! Feedbacks would be appreciated,{" "}
```
Comment (joeldenning) [inline src/root-config-guide/root-config-guide.component.js:166]:
```suggestion
          And that's all for now! Feedback is greatly appreciated,{" "}
```

File: src/root.component.js [hunk 33]
```diff
@@ -31,8 +31,9 @@ export default function Root(props) {
   }, []);
 
   useEffect(() => {
+    const directLocalStorage = localStorage.getItem("single-spa-playground");
     if (
-      !localStorage.getItem("single-spa-playground") &&
+      !(directLocalStorage && directLocalStorage.application) &&
```
Comment (joeldenning) [inline src/root.component.js:36]: `directLocalStorage` is a string, right? And strings won't have an `application` property on them. I think you need to JSON.parse() first?

File: src/instant-test/instant-test.component.js (hunk 8 above)
Comment (Sauloxd) [reply src/instant-test/instant-test.component.js:19]: Hmm I think I removed because since there is always only 1 application around, it will always override the last, so I thought there was no problem in removing this imperative reset! Hm... Now that I'm thinking, maybe I went overboard on deleting not-used-anymore code, and deprecated the multiple applications feature! Should I keep them, just in case this playground evolve again to support multiple apps?

File: src/introduction/introduction.component.js (hunk 25 above)
Comment (Sauloxd) [reply src/introduction/introduction.component.js:55]: Since it's one of the upsides of single-spa and microfrontends, helping others successfully deploy their apps seems to me like a valid next step! But not for now, this PR is so big that it can't fit anything else! sorry for making you read them top to bottom every time I make changes 😔

File: src/root-config-guide/root-config-guide.component.js [hunk 34]
```diff
@@ -0,0 +1,192 @@
...
+      const timeoutId = setTimeout(() => {
+        setCopying(false);
+      }, 500);
+
+      return () => clearTimeout(timeoutId);
+    }
+  }, [copying]);
+  const { application } = useContext(LocalStorageContext);
+  if (!application)
+    return <p>You need to register at least one application!</p>;
```
[the remark this answers is not in the document]
Comment (Sauloxd) [reply src/root-config-guide/root-config-guide.component.js:23]: Since the link is available in the sidebar, I just made this check so the playground won't explode when users try to explore the playground, before actually going through the steps

File: src/root-config-guide/root-config-guide.component.js (hunk 11 above)
Comment (Sauloxd) [reply src/root-config-guide/root-config-guide.component.js:175]: Nice!

File: src/shared/links.component.js (hunk 13 above)
Comment (Sauloxd) [reply src/shared/links.component.js:6]: Ok! I think there is a eslint that warns about this. Can I update the eslint-config with it? (eslintreact/jsx-no-target-blank)

File: src/shared/registered-app.component.js (hunk 14 above)
Comment (Sauloxd) [reply src/shared/registered-app.component.js:27]: Hmm, I think it doesn't really make sense... It doesn't put the user in a invalid state, because in places I need an application, I. added a check (like in root-config-guide page). Removing this feature, will definitely simply the code, should I?

File: src/verify-app-guide/verification-steps/application-dependencies.component.js (hunk 12 above)
Comment (Sauloxd) [reply src/verify-app-guide/verification-steps/application-dependencies.component.js:90]: It sure seem like a hack to me... I wondered if there was some API to get those dependencies (like, System.import() something, and ask System what dependencies are needed for a given specifier, but I found nothing in the API... so just to unblock me, I searched for the deps in the string haha
I learned about greed/lazy regex with you, so I could capture what's inside `[` until the first `]` 😁

File: src/verify-app-guide/verify-app-guide.component.js (hunk 17 above)
Comment (Sauloxd) [reply src/verify-app-guide/verify-app-guide.component.js:156]: oops

File: src/root.component.js (hunk 33 above)
Comment (Sauloxd) [reply src/root.component.js:36]: I will use the useLocalStorageData hook to get this data!

File: src/root-config-guide/root-config-guide.component.js (hunk 9 above)
Comment (Sauloxd) [reply src/root-config-guide/root-config-guide.component.js:87]: WOW just searched `lets vs let's` and I never knew "let's" was a contraction of `let us` ! hahahaha
I thought there were only contraction for `bleus has` `bleus is` to form "bleus's"

File: src/root-config-guide/root-config-guide.component.js (hunk 27 above)
Comment (Sauloxd) [reply src/root-config-guide/root-config-guide.component.js:82]: I meant it won't work just by `yarn start` ing as the create-single-spa react sibling tool, and the user should make some changes on it, before successfully being able to see his react single-spa application! How should I phrase this?

**Comment by Sauloxd** [comment]
@joeldenning updated this PR with your suggestions, and replied to some comments!
[New app to test]((https://back-prose.surge.sh)

Thanks for always carefully reviewing my PR, this one is quite big, and I'm starting to lose track myself of what's going on! hahaha Hopefully this will help more users to use single-spa!
Also, thanks for correcting my broken english 👍  I need to work on that too eheheh

File: src/instant-test/instant-test.component.js (hunk 8 above)
Comment (joeldenning) [reply src/instant-test/instant-test.component.js:19]: Ah good point. Yes this is unneeded now that there is only one app at a time. And no, I think only supporting one app at a time is the right way to go.

File: src/root-config-guide/root-config-guide.component.js (hunk 34 above)
[the remark this answers is not in the document]
Comment (joeldenning) [reply src/root-config-guide/root-config-guide.component.js]: Oh good point I didn't realize this was in the sidebar

File: src/verify-app-guide/verification-steps/application-dependencies.component.js (hunk 12 above)
Comment (joeldenning) [reply src/verify-app-guide/verification-steps/application-dependencies.component.js]: SystemJS does not allow for introspection of a module's dependencies. See the following links:

1. Dependencies are stored in the "load" object:
https://github.com/systemjs/systemjs/blob/c977894618c83928c0b843bf998f97bf2e0ffb1c/src/system-core.js#L184
2. The load object is stored in the SystemJS registry:
https://github.com/systemjs/systemjs/blob/c977894618c83928c0b843bf998f97bf2e0ffb1c/src/system-core.js#L167
3. The registry is a private Symbol, so it can't be accessed by code outside of SystemJS:
https://github.com/systemjs/systemjs/blob/c977894618c83928c0b843bf998f97bf2e0ffb1c/src/system-core.js#L22

File: src/shared/links.component.js (hunk 13 above)
Comment (joeldenning) [reply src/shared/links.component.js:6]: Sure!

File: src/shared/registered-app.component.js (hunk 14 above)
Comment (joeldenning) [reply src/shared/registered-app.component.js:27]: Yes, let's remove it

File: src/shared/links.component.js (hunk 13 above)
Comment (joeldenning) [reply src/shared/links.component.js:6]: Sure, could you actually add it to https://github.com/joeldenning/eslint-config-react-important-stuff? That eslint config is used all over the place and so adding it to that config would be even better

File: src/introduction/introduction.component.js (hunk 25 above)
Comment (joeldenning) [reply src/introduction/introduction.component.js]: I agree - that would be a great thing to add to the playground as a next step

File: src/root-config-guide/root-config-guide.component.js (hunk 27 above)
Comment (joeldenning) [reply src/root-config-guide/root-config-guide.component.js:87]: Maybe this?

> Running that create-single-spa comand will give you a root config, but it won't have your import map or applications set up yet. Here's what you need to do to get those set up:

File: src/introduction/introduction.component.js [hunk 35]
```diff
@@ -29,49 +34,54 @@ export default function Introduction(props) {
           applications.
         </section>
         <section>
-          The playground isn't the only way to use single-spa, but is one of the
-          easier ways. It will set you up with single-spa,{" "}
-          <a href="https://github.com/systemjs/systemjs">systemjs@6</a>, and{" "}
-          <a href="https://github.com/WICG/import-maps">import maps</a>. This
-          lets you do the following:
+          The playground will guide you to build a successful sing-e-spa
```
Comment (joeldenning) [inline src/introduction/introduction.component.js:37]:
```suggestion
          The playground will guide you to build a successful single-spa
```

File: src/introduction/introduction.component.js [hunk 36]
```diff
@@ -29,49 +34,54 @@ export default function Introduction(props) {
...
+          application, highlighting important steps, and giving you tools to
+          help you understand what went wrong. Mind that this isn't the only way
+          to use single-spa, but by better understanding the concepts you'll be
+          able to apply it in any way you like. Here's what we'll do:
           <ol>
-            <li>Have microfrontends that are in separate git repos.</li>
             <li>
-              Use different frameworks in the microfrontends, if you wish.
+              Have micro-frontends that are in separate git repos. We will be
```
Comment (joeldenning) [inline src/introduction/introduction.component.js:44]: Better to have consistent spelling of this I think. So to match other places:

```suggestion
              Have microfrontends that are in separate git repos. We will be
```

File: src/introduction/introduction.component.js [hunk 37]
```diff
@@ -29,49 +34,54 @@ export default function Introduction(props) {
...
+              focusing on React, but you could use Angular, Vue, etc
             </li>
-            <li>Deploy your microfrontends independently.</li>
+            <li>Test if your microfrontend is built up correctly</li>
```
Comment (joeldenning) [inline src/introduction/introduction.component.js:47]:
```suggestion
            <li>Test if your microfrontend is built correctly</li>
```

File: src/verify-app-guide/verification-steps/application-dependencies.component.js [hunk 38]
```diff
@@ -0,0 +1,122 @@
...
+              <div>Your app has the following shared dependencies:</div>
+              <CodeOutput code={sharedDependencies} />
+            </>
+          )}
+          {error && missingDeps.length > 0 ? (
+            <>
+              <div>Found missing dependencies:</div>
+              <CodeOutput code={missingDeps} />
+              <div>
+                Add them to the playground (using import-map-overrides on
```
Comment (joeldenning) [inline src/verify-app-guide/verification-steps/application-dependencies.component.js:56]: Could we automatically add these for them, via `importMapOverrides.addOverride('name', url)`? https://github.com/joeldenning/import-map-overrides/blob/master/docs/api.md#addoverride

Or is the point here to train them on how to use import-map-overrides? If so, that's fine with me.

File: src/verify-app-guide/verification-steps/application-dependencies.component.js (hunk 12 above)
Comment (Sauloxd) [reply src/verify-app-guide/verification-steps/application-dependencies.component.js]: I see... thanks!

File: src/root-config-guide/root-config-guide.component.js (hunk 27 above)
Comment (Sauloxd) [reply src/root-config-guide/root-config-guide.component.js:87]: Sounds good to me

File: src/verify-app-guide/verification-steps/application-dependencies.component.js (hunk 38 above)
Comment (Sauloxd) [reply src/verify-app-guide/verification-steps/application-dependencies.component.js:56]: I think if we automatically add them, the user might miss this chance of learning how to handle dependencies... but if we don't want the user to experiment with `import-map-overrides`, we can add a button to add for them, hiding the import-map-overrides layer.
Having them manually press the button, may give time for them to read what is actually happening
wdyt? maybe I can do this is a next PR, cause this one is kinda bloated already haha

**Comment by Sauloxd** [comment]
Updated:
- Update with suggested copies
- Remove delete button when editing registered app

new app link: https://ubiquitous-frogs.surge.sh/

File: src/verify-app-guide/verification-steps/application-dependencies.component.js (hunk 38 above)
Comment (joeldenning) [reply src/verify-app-guide/verification-steps/application-dependencies.component.js:56]: Yeah I think teaching them to use import-map-overrides is important. I'm okay with not doing it in this PR or ever
